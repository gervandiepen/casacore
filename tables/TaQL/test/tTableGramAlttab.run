#!/bin/sh
#-----------------------------------------------------------------------------
# Script to test the TableGram and TableParse class.  All files generated will
# be deleted on exit.
#=============================================================================

# Create a symlink to achieve that casacore_memcheck (if used) will use
# that name 
rm -f tTableGramAlttab
ln -s tTableGram tTableGramAlttab


# Execute all kind of table creation commands.
# Check the result using showtable (leave out version line and path).
curdir=`pwd`

# First create the tables to use.
echo ""
$casa_checktool ./tTableGram 'create table tTableGramAlttab_tmp.tab2 (col1 i4 [shape=[2,3], unit="m"], col2 B) dminfo [TYPE="IncrementalStMan",NAME="ISM1",SPEC=[BUCKETSIZE=16384],COLUMNS=["col1"]]'
$casa_checktool ./tTableGram 'create table tTableGramAlttab_tmp.tab2/subtab'
../../apps/showtable in=tTableGramAlttab_tmp.tab2 | grep -v 'showtable:' | sed -e "s%$curdir/%%"
$casa_checktool ./tTableGram 'alter table tTableGramAlttab_tmp.tab2/subtab ADD COLUMN colxyz S SET KEYWORD colxyz::skey="newval"'
../../apps/showtable in=tTableGramAlttab_tmp.tab2/subtab colkey=T | grep -v 'showtable:' | sed -e "s%$curdir/%%"

echo ""
$casa_checktool ./tTableGram 'alter table tTableGramAlttab_tmp.tab2 SET KEYWORD subtab="Table: tTableGramAlttab_tmp.tab2/subtab", ::key1=3, col1::subrec=[=], col1::subrec.k1="3x"'
../../apps/showtable in=tTableGramAlttab_tmp.tab2 tabkey=T colkey=T | grep -v 'showtable:' | sed -e "s%$curdir/%%"

echo ""
$casa_checktool ./tTableGram 'alter table tTableGramAlttab_tmp.tab2 FROM ::subtab t1 SET KEYWORD col1::subrec.k1 AS t1.colxyz::skey, RENAME KEYWORDS key1 TO key1n DROp keyword subtab'
../../apps/showtable in=tTableGramAlttab_tmp.tab2 tabkey=T colkey=T | grep -v 'showtable:' | sed -e "s%$curdir/%%"

echo ""
$casa_checktool ./tTableGram 'alter table tTableGramAlttab_tmp.tab2 ADD COLUMN col1a i4 [shape=[2,3], unit="m", dmtype="IncrementalStMan"], col2a B'
../../apps/showtable in=tTableGramAlttab_tmp.tab2 | grep -v 'showtable:' | sed -e "s%$curdir/%%"

echo ""
$casa_checktool ./tTableGram 'alter table tTableGramAlttab_tmp.tab2 RENAME COLUMN col1a to col1b, col2a to col2b'
../../apps/showtable in=tTableGramAlttab_tmp.tab2 | grep -v 'showtable:' | sed -e "s%$curdir/%%"

$casa_checktool ./tTableGram 'alter table tTableGramAlttab_tmp.tab2 DROP COLUMN col1b,col2 ,ADDrows 4 +5'
../../apps/showtable in=tTableGramAlttab_tmp.tab2 | grep -v 'showtable:' | sed -e "s%$curdir/%%"
 
exit 0
