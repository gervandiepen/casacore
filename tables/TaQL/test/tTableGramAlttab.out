
create table tTableGramAlttab_tmp.tab2 (col1 i4 [shape=[2,3], unit="m"], col2 B) dminfo [TYPE="IncrementalStMan",NAME="ISM1",SPEC=[BUCKETSIZE=16384],COLUMNS=["col1"]]
    has been executed
    cretab result of 0 rows
2 selected columns:  col1 col2
create table tTableGramAlttab_tmp.tab2/subtab
    has been executed
    cretab result of 0 rows
0 selected columns: 

Structure of table tTableGramAlttab_tmp.tab2
------------------ 
0 rows, 2 columns in little endian format (using 2 data managers)

 IncrementalStMan file=table.f0  name=ISM1  bucketsize=16384
    ActualCacheSize=1 PERSCACHESIZE=1
  col1 Int      shape=[2,3] unit=[m]

 StandardStMan file=table.f1  name=StandardStMan  bucketsize=128
    ActualCacheSize=2 PERSCACHESIZE=2 IndexLength=118
  col2 Bool     scalar
alter table tTableGramAlttab_tmp.tab2/subtab ADD COLUMN colxyz S SET KEYWORD colxyz::skey="newval"
    has been executed
    alttab result of 0 rows
0 selected columns: 

Structure of table tTableGramAlttab_tmp.tab2/subtab
------------------ 
0 rows, 1 columns in little endian format (using 1 data managers)

 StandardStMan file=table.f0  name=SSM  bucketsize=384
    ActualCacheSize=2 PERSCACHESIZE=2 IndexLength=118
  colxyz String   scalar

Keywords of main table 
----------------------
  Column colxyz
    skey: String "newval"


alter table tTableGramAlttab_tmp.tab2 SET KEYWORD subtab="Table: tTableGramAlttab_tmp.tab2/subtab", ::key1=3, col1::subrec=[=], col1::subrec.k1="3x"
    has been executed
    alttab result of 0 rows
0 selected columns: 

Structure of table tTableGramAlttab_tmp.tab2
------------------ 
0 rows, 2 columns in little endian format (using 2 data managers)

 IncrementalStMan file=table.f0  name=ISM1  bucketsize=16384
    ActualCacheSize=1 PERSCACHESIZE=1
  col1 Int      shape=[2,3] unit=[m]

 StandardStMan file=table.f1  name=StandardStMan  bucketsize=128
    ActualCacheSize=2 PERSCACHESIZE=2 IndexLength=118
  col2 Bool     scalar

 SubTables:
    tTableGramAlttab_tmp.tab2/subtab

Keywords of main table 
----------------------
  Table Keywords
    subtab: Table tTableGramAlttab_tmp.tab2/subtab
    key1: Int 3

  Column col1
    QuantumUnits: String array with shape [1]
      [m]
    subrec: {
      k1: String "3x"
    }


alter table tTableGramAlttab_tmp.tab2 FROM ::subtab t1 SET KEYWORD col1::subrec.k1 AS t1.colxyz::skey, RENAME KEYWORDS key1 TO key1n DROp keyword subtab
    has been executed
    alttab result of 0 rows
0 selected columns: 

Structure of table tTableGramAlttab_tmp.tab2
------------------ 
0 rows, 2 columns in little endian format (using 2 data managers)

 IncrementalStMan file=table.f0  name=ISM1  bucketsize=16384
    ActualCacheSize=1 PERSCACHESIZE=1
  col1 Int      shape=[2,3] unit=[m]

 StandardStMan file=table.f1  name=StandardStMan  bucketsize=128
    ActualCacheSize=2 PERSCACHESIZE=2 IndexLength=118
  col2 Bool     scalar

Keywords of main table 
----------------------
  Table Keywords
    key1n: Int 3

  Column col1
    QuantumUnits: String array with shape [1]
      [m]
    subrec: {
      k1: String "newval"
    }


alter table tTableGramAlttab_tmp.tab2 ADD COLUMN col1a i4 [shape=[2,3], unit="m", dmtype="IncrementalStMan"], col2a B
    has been executed
    alttab result of 0 rows
0 selected columns: 

Structure of table tTableGramAlttab_tmp.tab2
------------------ 
0 rows, 4 columns in little endian format (using 3 data managers)

 IncrementalStMan file=table.f0  name=ISM1  bucketsize=16384
    ActualCacheSize=1 PERSCACHESIZE=1
  col1  Int      shape=[2,3] unit=[m]

 StandardStMan file=table.f1  name=StandardStMan  bucketsize=128
    ActualCacheSize=2 PERSCACHESIZE=2 IndexLength=118
  col2  Bool     scalar

 StandardStMan file=table.f2  name=SSM  bucketsize=260
    ActualCacheSize=2 PERSCACHESIZE=2 IndexLength=118
  col1a Int      shape=[2,3] unit=[m]
  col2a Bool     scalar

alter table tTableGramAlttab_tmp.tab2 RENAME COLUMN col1a to col1b, col2a to col2b
    has been executed
    alttab result of 0 rows
0 selected columns: 

Structure of table tTableGramAlttab_tmp.tab2
------------------ 
0 rows, 4 columns in little endian format (using 3 data managers)

 IncrementalStMan file=table.f0  name=ISM1  bucketsize=16384
    ActualCacheSize=1 PERSCACHESIZE=1
  col1  Int      shape=[2,3] unit=[m]

 StandardStMan file=table.f1  name=StandardStMan  bucketsize=128
    ActualCacheSize=2 PERSCACHESIZE=2 IndexLength=118
  col2  Bool     scalar

 StandardStMan file=table.f2  name=SSM  bucketsize=260
    ActualCacheSize=2 PERSCACHESIZE=2 IndexLength=118
  col1b Int      shape=[2,3] unit=[m]
  col2b Bool     scalar
alter table tTableGramAlttab_tmp.tab2 DROP COLUMN col1b,col2 ,ADDrows 4 +5
    has been executed
    alttab result of 9 rows
0 selected columns: 

Structure of table tTableGramAlttab_tmp.tab2
------------------ 
9 rows, 2 columns in little endian format (using 2 data managers)

 IncrementalStMan file=table.f0  name=ISM1  bucketsize=16384
    ActualCacheSize=1 PERSCACHESIZE=1
  col1  Int      shape=[2,3] unit=[m]

 StandardStMan file=table.f2  name=SSM  bucketsize=260
    ActualCacheSize=2 PERSCACHESIZE=2 IndexLength=134
  col2b Bool     scalar
